### PR TITLE
Point dev instance at AWS sandbox rather than parse.com (BL-3726)

### PR DESCRIPTION
**Pull request by andrew-polk**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomlibrary/250)

### CODE REVIEW

File: src/app/modules/common/services.js [hunk 1]
```diff
@@ -184,6 +187,9 @@ angular.module('BloomLibraryApp.services', ['restangular'])
 
     .service('sharedService', function() {
         this.isPublicSite = window.location.host.indexOf("bloomlibrary.org") === 0;
+        
+        this.prodUrl = "https://api.parse.com/1"; // 1/indicates rev 1 of parse.com API
```
Comment (hatton) [inline src/app/modules/common/services.js]: I'm a real anti-fan of abbreviations :-)

File: src/app/modules/common/services.js [hunk 2]
```diff
@@ -65,16 +65,19 @@ angular.module('BloomLibraryApp.services', ['restangular'])
...
+        } else {
             // we're live! Use the real silbloomlibrary api strings.
             headers = {
                 'X-Parse-Application-Id': 'R6qNTeumQXjJCMutAJYAwPtip1qBulkFyLefkCE5',
                 'X-Parse-REST-API-Key': 'P6dtPT5Hg8PmBCOxhyN9SPmaJ8W4DcckyW0EZkIx'
             };
         }
 		restangularConfig = function (restangularConfigurer) {
-			restangularConfigurer.setBaseUrl('https://api.parse.com/1'); // 1/indicates rev 1 of parse.com API
+			if (!sharedService.isPublicSite) {
```
Comment (hatton) [inline src/app/modules/common/services.js]: Public --> Production ?

File: src/app/modules/common/services.js [hunk 3]
```diff
@@ -446,24 +453,6 @@ angular.module('BloomLibraryApp.services', ['restangular'])
 		// By that time books will be an array of json-encoded book objects from parse.com.
 		this.getFilteredBookRange = function (first, count, searchString, shelfKey, lang, tag, allLicenses, sortBy, ascending, includeOutOfCirculation) {
 			var defer = $q.defer(); // used to implement angularjs-style promise
-            var fixLangPtrs = function(book) {
```
Comment (hatton) [inline src/app/modules/common/services.js]: So this is probably just a function that isn't used anymore, so you removed it?

**Comment by hatton** [comment]
@JohnThomson Can you review? Thx.

**Comment by JohnThomson** [comment]
I tend to agree with JohnH's suggestions. I can't verify the new URLs but presume you've checked, and also that the new version of parse doesn't cause obvious problems. I think it's OK to use a private function...if it goes away or changes it should quickly become obvious. Conceivably you could make it more so by somewhere in startup using it once and checking the result and failing really obviously if it no longer does what's expected. But I'm not sure it's worth it.

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/app/modules/common/services.js, line 466 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250#-KQNYPwkMtEgBeTycwBD:-KQNYPwkMtEgBeTycwBE:bg9ronl) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/3d92e3a51e67c4253ee4fa12b33b0923735b263f/src/app/modules/common/services.js#L466)):*
> ```JavaScript
>                     book.set("langPointers", fixedArray);
>                 }
>             };
> ```

Is this no longer needed because of moving to a later version of parse? It was definitely necessary at one point. The thing to check for would, I think, be that we're correctly displaying the languages of a book in the browse and detail views, and successfully filtering by language.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250#-:-KQNbJp86CbpkmkXPiVd:b-1erfrd)*

**Comment by andrew-polk** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/app/modules/common/services.js, line 76 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250#-KQLYpCiFxRMwWJSxV3n-r1-76:-KQXDcg8DcTLvhwmFDAG:b-80tlqk) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/66a02b61af7af53c90a2728d073219aa4cc6448d/src/app/modules/common/services.js#L76)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary>
> Public --> Production ?
</details>
Sure.

---

*[src/app/modules/common/services.js, line 191 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250#-KQLYpCiFxRMwWJSxV3n-r1-191:-KQXDeAm72cDlIZycKaL:b-jngnyz) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/66a02b61af7af53c90a2728d073219aa4cc6448d/src/app/modules/common/services.js#L191)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary>
> I'm a real anti-fan of abbreviations :-)
</details>
Ok, will change.

---

*[src/app/modules/common/services.js, line 466 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250#-KQNYPwkMtEgBeTycwBD:-KQXCo_J15ghFfg6wJ3H:bcxtp4m) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/3d92e3a51e67c4253ee4fa12b33b0923735b263f/src/app/modules/common/services.js#L466)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary>
> Is this no longer needed because of moving to a later version of parse? It was definitely necessary at one point. The thing to check for would, I think, be that we're correctly displaying the languages of a book in the browse and detail views, and successfully filtering by language.
</details>
I could not find any information online whatsoever regarding breaking changes parse has made to the API both before and after the migration from parse.com to parse server.

I had to update the API version to at least 1.6.14 which removed the toJSON method. In researching, I found _toFullJSON which seems to replace the need for this code. So I have no idea if _toFullJSON was there before and we missed it, or if it was added.

I did test everything I knew to test regarding languages (and everything else) and don't see any problems.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250)*

**Comment by andrew-polk** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/app/modules/common/services.js, line 76 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250#-KQLYpCiFxRMwWJSxV3n-r1-76:-KQXFtlWM2yjPLQWCF1I:b-896fix) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/66a02b61af7af53c90a2728d073219aa4cc6448d/src/app/modules/common/services.js#L76)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary>
> Sure.
</details>
Done.

---

*[src/app/modules/common/services.js, line 191 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250#-KQLYpCiFxRMwWJSxV3n-r1-191:-KQXFuD7jK4U1e_jOXVd:b-896fix) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/66a02b61af7af53c90a2728d073219aa4cc6448d/src/app/modules/common/services.js#L191)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary>
> Ok, will change.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250)*

**Comment by JohnThomson** [comment]
:lgtm:

---

Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/app/modules/common/services.js, line 466 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250#-KQNYPwkMtEgBeTycwBD:-KQ_y2NeXP1K3mGX3FUX:b-gxztsp) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/3d92e3a51e67c4253ee4fa12b33b0923735b263f/src/app/modules/common/services.js#L466)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary>
> I could not find any information online whatsoever regarding breaking changes parse has made to the API both before and after the migration from parse.com to parse server.
> 
> I had to update the API version to at least 1.6.14 which removed the toJSON method. In researching, I found _toFullJSON which seems to replace the need for this code. So I have no idea if _toFullJSON was there before and we missed it, or if it was added.
> 
> I did test everything I knew to test regarding languages (and everything else) and don't see any problems.
</details>
Fair enough.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250#-:-KQ_y7kfno-vp7umciRP:bnfp4nl)*

**Comment by JohnThomson** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/250)*

**Pull request merged**